ocaml/typing/typedtree.ml

```diff
@@ -909,12 +909,13 @@ let iter_pattern_full ~both_sides_of_or f sort pat =
           List.iter (fun (_, lbl, pat) ->
             (loop f) (Jkind.sort_of_jkind lbl.lbl_jkind) pat)
             lbl_pat_list
+      | Tpat_array (_, arg_sort, patl) ->
+        List.iter (loop f arg_sort) patl
       (* Cases where the inner things must be value: *)
       | Tpat_variant (_, pat, _) -> Option.iter (loop f Jkind.Sort.value) pat
       | Tpat_tuple patl -> List.iter (loop f Jkind.Sort.value) patl
         (* CR layouts v5: tuple case to change when we allow non-values in
            tuples *)
-      | Tpat_array (_, patl) -> List.iter (loop f Jkind.Sort.value) patl
```

Comment:
Why move this? That will cause more challenge when merging later.

Reply:
Moved it to try to keep the `(* Cases where the inner things must be value: *)` and `(* Cases where we compute the sort of the inner thing from the pattern *)` comments true.

I agree it may not be worth it and can introduce merge conflicts. Reverted the change in a new commit